confidant/authnz/__init__.py

```diff
@@ -237,6 +247,51 @@ def decorated(*args, **kwargs):
     return decorated
 
 
+def require_role(role):
+    """
+    Get the users role in credential metadata
+    """
+    def decorated(f):
+        @wraps(f)
+        def wrapper(*args, **kwargs):
+            cred_id = kwargs['id']
+            cred = Credential.get(cred_id)
+
+            if app.config['USE_ROLES']:
+                try:
+                    user_role = user_mod.current_role()
+                    role_name_rw = app.config['ROLE_RW_NAME'].lower()
+                    role_name_r = app.config['ROLE_RO_NAME'].lower()
+                    groups_rw = cred.metadata.get(role_name_rw)
+                    groups_r = cred.metadata.get(role_name_r)
```

Comment:
Should each of these be `.lower()` as well? Pretty sure we don't force these to lowercase in the model.

Reply:
fixed below